### PR TITLE
Drunken Brawlening: Intoxication alters your unarmed effectiveness, bartenders are stronger fighters while drunk, deterministic stagger combo is back (your average assistant doesn't get knockdowns)

### PR DESCRIPTION
**Pull request by necromanceranne**
### Intoxication and Punching

Being drunk juuuust enough allows you to land stronger punches and take punches a little more easily.

However, being TOO drunk makes you WORSE at these things, and makes your unarmed accuracy an unarmed PENALTY. Don't get drunk if you can't hold your liquor.

However, fighting drunk makes you really sick, so be careful as to not suddenly start vomiting everywhere.

### Drunken Brawler

Bartenders are now drunken brawlers. They gain additional punching prowess if they're ever intoxicated, it does not matter how much nor do they ever suffer a penalty as a result of being TOO drunk.

The more damage the bartender has taken, the stronger they hit back. (added by recommendation from WalterTruck) I have to mess the scaling on this more.

### Stagger Combo/Unarmed Combos

The stagger combo is back, but significantly less stun heavy except on the extreme ends of unarmed upgrading.

When your opponent has been sufficiently damaged enough, staggered and doesn't have any riot protection gear on, you can inflict additional status effects when you punch an opponent.

The necessary value for this effect is: Half the target's Brute and Burn damage combined exceeds a value of: Limb accuracy + armor block (min. 40. max 200)

While the thresholds are more deterministic, the effects are random. The better you are at punching, the better the effect, but the higher your targets armor, the weaker the effect. All effects are positive for the attacker and negative for the target.

The dice roll is such; (a roll between -20 to 20) + limb accuracy - armor block

| Stage/Value | Effect |
| ------------- | ------------- |
| Stage 1 (-Infinity to 0  | Staggered for 1 second |
| Stage 2 (1 to 10)  | Eye Blur for 5 seconds |
| Stage 3 (11 to 30)  | Eye Blur, Dizzying, Confusion for 5 seconds |
| Stage 4 (31 to 40)  | Blindness, Dizzying, Confusion for 5 seconds |
| Stage 5 (41 to 45)  | Knockdown for 4 seconds |
| Stage 6 (46 to Infinity)  | Knockdown for 4 seconds, 5 Brute with huge wound chance |

Stages 5 and 6 are not possible to reach for roundstart humans who aren't intoxicated. It is almost impossible to get high effects against someone who is in full security gear as a roundstart human

### Minor Changes

Any unarmed related thresholds or values that would use Stamina damage now use half of the total of Burn and Brute damage on the relevant individual. This affects punch accuracy, the threshold for getting a stagger combo, and the threshold for someone being more vulnerable to grabs.

## Why It's Good For The Game

I just really wanted to add some drunken brawling/drunk fist style martial art stuff because I thought it would be fun, and bar brawls are where most people start fist fights. It's thematically entertaining.

The bartender, by extension, should be pretty proficient at this kind of brawling. It also works out great when working with chefs to beat people up with their fists in their own, more unrefined style.

The situation with the unarmed knockdown/stamina pr was kind of a mess, since at the time I was already going to take it out after discussing things with Walter and we were working out some alternatives together. But uh...well, it wasn't to be. Particularly, there was a lot of stuff in unarmed fighting that used Stamina damage that didn't actually get removed at all. So this cleans that up.

I said I'd find a good alternative down the line to fill that gap, and this is it. For your run-of-the-mill fistfights, there are no random knockdowns. It's only when you are actually trying to brawl with a cybernetically enhanced human that this kind of thing starts to crop up. Or your opponent is a golem. Or you are both really drunk (and thus everyone starts vomiting because they're getting beaten up)

The average assistant beating up another assistant is at best going to inflict some eye blurring and dizziness, assuming the fight even lasts that long. But this should be sufficient to create space to escape, or even meaningfully disable someone who might simply be invading your department. It is much less helpful against a more sufficiently geared opponent. 

The main reason for focusing on this is to give the average joe some kind of way for a heel turn in the event they're having to scuffle with departmental invaders or escape some determined opponent. It does not, however, allow them to manhandle a nuclear operative in an elite suit while our would-be hero is entirely butt-naked and drunk. Not that they shouldn't try, that's funny as hell.

Thus, the effects chosen are more useful for escaping and befuddlement than necessarily stopping the fight outright. If you have improved arms, your arms are weapons and start to act like them, so you start seeing more significant results.

It is also a nice thematic tie to unarmed's zany cousin; tackling, since the randomized effects are borrowing the logic form there, and it is cool to have those tied a bit closer.

## Changelog
:cl:
add: Drunken fist fighting now has bonuses and penalties based on how intoxicated you are. Controlled liquor intake could make you a better brawler. Though you might vomit if you go too hard.
add: Bartenders are now Drunken Brawlers. If they're drunk at all (no matter how drunk), they're stronger at fist fighting.
add: Reintroduces the deterministic stagger combo. The effects of the combo are more varied, and based on attacker unarmed effectiveness and the defender's armor. Read the PR for more details. It's complicated.
qol: Unarmed effects that would utilize stamina values now use a split of half brute, half burn to determine outcomes or for meeting thresholds. This affects; punch accuracy, stagger combo thresholds and grab vulnerability thresholds.
/:cl:

### CODE REVIEW

**Comment by DrAmazing343** [comment]
Every time I see one of your PR's, my face lights up. Yet ANOTHER absolute Anne Victory here- Especially since this is making me excited to go back to my old main of Bartender. Absolute cinema.

**Comment by necromanceranne** [comment]
The biggest add here is that ethereals are not uncharacteristically shit at unarmed fighting for absolutely no reason, simply because they do exclusively burn damage and none of the relevant procs acknowledge burn whatsoever.

**Comment by Loafin34** [comment]
what is drunken brawling tied to? is it the bartender's brain itself, is it a skillchip like chefqc, is it a liver trait?

**Comment by necromanceranne** [comment]
It's a mind trait rn Lemme make it a skillchip real quick.

**Comment by SomeoneYouProbablyKnow** [comment]
Aren't bartenders supposed to NOT be drinking? Like the reliably sober person at the bar? The drunken fist thing is funny but pushing bartender towards being a drunkard feels weird.
Sadly can't slap that effect on the most appropriate role because assistants are assistants and don't deserve any unique buffs.

The mechanics themselves are cool, i'm not a fan of the combo-y combat but that's literally a me issue.

**Comment by necromanceranne** [comment]
If you worked at a Nanotrasen bar, would you want to be sober?

**Comment by SomeoneYouProbablyKnow** [comment]
No, and that's the same reason a bartender has a suit and not regular NTcoveralls, there's a certain fantasy in the job even if most of it's lost on people. And honestly a fancy bartender on a shithole station is more interesting than everything is a corpo dystopia. Contrast is fun.

**Comment by necromanceranne** [comment]
Well, for what it is worth,  you do not necessarily _have_ to use your full drunken power. It's not like it is mandatory, just kind of there to be used. Like your shotgun. Choosing to not use something is as equally an interesting narrative element.

**Comment by Loafin34** [comment]
i think the drunken brawler should be a buyable skillchip because it'd be funny to imagine a powergamer intentionally getting shitfaced and pulling up on security with strongarm, advanced limbs, gravity-unlocked herculean core, and master athletics

**Comment by nianjiilical** [comment]
> Aren't bartenders supposed to NOT be drinking? Like the reliably sober person at the bar? The drunken fist thing is funny but pushing bartender towards being a drunkard feels weird. Sadly can't slap that effect on the most appropriate role because assistants are assistants and don't deserve any unique buffs.
> 
> The mechanics themselves are cool, i'm not a fan of the combo-y combat but that's literally a me issue.

counterargument: i love the mental image of the bartender seeing an overly aggressive greyshirt ruining their bar, rolling their sleeves up, taking a swig of pan-galactic gargle blaster like jamie streetfighter and unleashing concentrated ass kicking

**Comment by SomeoneYouProbablyKnow** [comment]
> Well, for what it is worth, you do not necessarily _have_ to use your full drunken power. It's not like it is mandatory, just kind of there to be used. Like your shotgun. Choosing to not use something is as equally an interesting narrative element.

Yes but you know well enough that's not the same not least of all because people will gravitate towards this thing like they gravitate to playing chef a certain way because of cqc. But not much to be done, the feature is real funny. Frankly i just like complaining.

> counterargument: i love the mental image of the bartender seeing an overly aggressive greyshirt ruining their bar, rolling their sleeves up, taking a swig of pan-galactic gargle blaster like jamie streetfighter and unleashing concentrated ass kicking

Undoubtedly funny, but i kinda feel like that's already the vibe of half the station and when everyone is rolling up their sleeves and unleasing ass kickings with their trademark *thing* all the time you lose out on the contrast.

**Comment by Sakamoto4ka** [comment]
Drunk martial art from curse 220

**Comment by EricZilla** [comment]
The fact that bartender/chef conflicts have been upgraded to full on martial arts duels

**Comment by atlasle** [comment]
bartenders already have a shotgun as a combat edge they don't need something that they can always use as long as there's alcohol around, people are gonna start picking bartender to powergame now

**Comment by TetrakoHateThis** [comment]
At least there's signs unarmed combat might be getting more interesting and less 'last resort RNG gamble'. Now, if only investing into Athletics made ungloved combat good.

**Comment by DrAmazing343** [comment]
The Shotgun is a coward's weapon, good for nothing more than a vague "shoo" out of the bar area. You know why? Because for a civilian, it DOES miss. The only thing it's really good for is typebaiting a two-tap stun on someone and that's LAME.

I'm super excited for that "rolling up sleeves, taking a shot" sort-of throwing someone out of my bar, because that fits my character quite excellently. I can understand thematic concerns... to a point. Your character and their themes are not emblematic to those of the entire Station or the entire community, so I don't see the point in insisting that the Bartender is the "one sober man" on this shitheap Station. In contrast, I think in the wake of this PR those themes will have MORE value BECAUSE they're going to be rarer(?)

**Comment by atlasle** [comment]
> The Shotgun is a coward's weapon, good for nothing more than a vague "shoo" out of the bar area. You know why? Because for a civilian, it DOES miss. The only thing it's really good for is typebaiting a two-tap stun on someone and that's LAME.
> 
> 
> 
> I'm super excited for that "rolling up sleeves, taking a shot" sort-of throwing someone out of my bar, because that fits my character quite excellently. I can understand thematic concerns... to a point. Your character and their themes are not emblematic to those of the entire Station or the entire community, so I don't see the point in insisting that the Bartender is the "one sober man" on this shitheap Station. In contrast, I think in the wake of this PR those themes will have MORE value BECAUSE they're going to be rarer(?)

i mean sure but we kinda have to do something about the shotgun still, maybe it could just be a trophy on the wall? (still functional just initially decorative, bartender's ammo would be moved somewhere else then)

**Comment by SomeoneYouProbablyKnow** [comment]
Why would you have to do anything about the shotgun? You can still use the shotgun, frankly it's funnier and more fitting for a drunk kunfu bartender to have the shotgun.

**Comment by Eronymun** [comment]
> > The Shotgun is a coward's weapon, good for nothing more than a vague "shoo" out of the bar area. You know why? Because for a civilian, it DOES miss. The only thing it's really good for is typebaiting a two-tap stun on someone and that's LAME.
> > I'm super excited for that "rolling up sleeves, taking a shot" sort-of throwing someone out of my bar, because that fits my character quite excellently. I can understand thematic concerns... to a point. Your character and their themes are not emblematic to those of the entire Station or the entire community, so I don't see the point in insisting that the Bartender is the "one sober man" on this shitheap Station. In contrast, I think in the wake of this PR those themes will have MORE value BECAUSE they're going to be rarer(?)
> 
> i mean sure but we kinda have to do something about the shotgun still, maybe it could just be a trophy on the wall? (still functional just initially decorative, bartender's ammo would be moved somewhere else then)

Tbh this is looking like it's not going to even in the same galaxy in terms of the strength that CQC has, so the shotgun is probably still fine.

File: code/modules/mob/living/carbon/human/_species.dm [hunk 1]
```diff
@@ -1034,14 +1034,32 @@ GLOBAL_LIST_EMPTY(features_by_species)
...
+	else if(user_drunkenness >= 60)
+		limb_accuracy = -limb_accuracy // good luck landing a punch now, you drunk fuck
+		user.adjust_disgust(5)
+
 	var/obj/item/bodypart/affecting = target.get_bodypart(target.get_random_valid_zone(user.zone_selected))
 
 	var/miss_chance = 100//calculate the odds that a punch misses entirely. considers stamina and brute damage of the puncher. punches miss by default to prevent weird cases
 	if(attacking_bodypart.unarmed_damage_low)
-		if((target.body_position == LYING_DOWN) || HAS_TRAIT(user, TRAIT_PERFECT_ATTACKER) || staggered) //kicks and attacks against staggered targets never miss (provided your species deals more than 0 damage)
+		if((target.body_position == LYING_DOWN) || HAS_TRAIT(user, TRAIT_PERFECT_ATTACKER) || staggered || user_drunkenness && HAS_TRAIT(user, TRAIT_DRUNKEN_BRAWLER)) //kicks and attacks against staggered targets never miss (provided your species deals more than 0 damage) //drunken brawlers while drunk also don't miss
```
Comment (Ghommie) [inline code/modules/mob/living/carbon/human/_species.dm:1059]:
```suggestion
		if((target.body_position == LYING_DOWN) || HAS_TRAIT(user, TRAIT_PERFECT_ATTACKER) || staggered || (user_drunkenness && HAS_TRAIT(user, TRAIT_DRUNKEN_BRAWLER))) //kicks and attacks against staggered targets never miss (provided your species deals more than 0 damage). Drunken brawlers while drunk also don't miss
```

File: code/modules/mob/living/carbon/human/_species.dm [hunk 2]
```diff
@@ -1080,6 +1112,65 @@ GLOBAL_LIST_EMPTY(features_by_species)
 			target.force_say()
 		log_combat(user, target, "punched")
 
+	// If our target is staggered and has sustained enough damage, we can apply a randomly determined status effect to inflict when we punch them.
+	// The effects are based on the punching effectiveness of our attacker. Some effects are not reachable by the average human, and require augmentation to reach or being a species with a heavy punch effectiveness.
+	// Or they're just drunk enough.
+	if(HAS_TRAIT(src, TRAIT_BRAWLING_KNOCKDOWN_BLOCKED) && target.stat == DEAD) //If our target is dead or has specailized armor, there is no way to inflict these effects.
```
Comment (Ghommie) [inline code/modules/mob/living/carbon/human/_species.dm:1118]:
```suggestion
	if(HAS_TRAIT(target, TRAIT_BRAWLING_KNOCKDOWN_BLOCKED) || target.stat == DEAD) //If our target is dead or has specailized armor, there is no way to inflict these effects.
```

File: code/modules/mob/living/carbon/human/_species.dm [hunk 3]
```diff
@@ -1080,6 +1112,65 @@ GLOBAL_LIST_EMPTY(features_by_species)
...
+			target.visible_message(span_danger("[user]'s [atk_verb] hits [target] so hard, their eyes water! Ouch!"), \
+				span_danger("You are hit viciously by [user]'s [atk_verb], and your eyes begin to water!"), span_hear("You hear a thud!"), COMBAT_MESSAGE_RANGE, user)
+			to_chat(user, span_warning("Your [atk_verb] causes [target] to tear up!"))
+
+		if (11 to 30)
+			target.adjust_dizzy_up_to(5 SECONDS, 10 SECONDS)
+			target.adjust_eye_blur_up_to(5 SECONDS, 10 SECONDS)
+			target.adjust_confusion_up_to(5 SECONDS, 10 SECONDS)
+			target.visible_message(span_danger("[user]'s [atk_verb] hits [target] so hard, they are sent reeling in agony! Damn!"), \
+				span_danger("You are hit viciously by [user]'s [atk_verb], and everything becomes a dizzying blur!"), span_hear("You hear a thud!"), COMBAT_MESSAGE_RANGE, user)
```
Comment (Ghommie) [inline code/modules/mob/living/carbon/human/_species.dm:1148]: I don't think I'd have the time to read all of this during combat, especially if the font is too big. This is an issue shared with many other combat-related messages tho. Also it kinda sucks to receive several messages from a single punch ngl.

**Review by Ghommie — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/carbon/human/_species.dm (hunk 3 above)
Comment (necromanceranne) [reply code/modules/mob/living/carbon/human/_species.dm:1148]: I'm just worried that balloon text might seem a little...gamey? Not sure how else to put it.

Comment (necromanceranne) [reply code/modules/mob/living/carbon/human/_species.dm:1148]: I'll swap it to span_warning so it is smaller at least

**Comment by Ghommie** [comment]
Merge conflicts.

**Comment by necromanceranne** [comment]
oooh I'm a victim of the sporadic test failures

**Pull request merged**